### DISCUSSION TITLE
how to install oc-bibtex-actions with straight.el?

### DISCUSSION THREAD

**Discussion by gwbrck** [emacs-citar/citar#275] [Q&A] [answered]
Obviously you can't just install oc-bibtex-actions with straight. Since oc-bibtex-actions is not a separate package on Melpa. 
What would be the "correct" way to proceed? 

First load the package with `(use-package bibtex-actions)` and then load the oc-bibtex.el under `:config ` with `(require ' oc-bibtex-actions)`?

I appreciate your help! 

Thank you!

**Comment by bdarcus** [comment]
I'm not an expert on use-package, but in the README, I have a separate use-package statement for oc-bibtex-actions:

https://github.com/bdarcus/bibtex-actions#org-cite

So you'd have:

```elisp
(use-package bibtex-actions ...)
(use-package oc-bibtex-actions ...)
```

But probably the first isn't needed, since `oc-bibtex-actions` imports `bibtex-actions`.

Does that work?

**Comment by gwbrck** [comment]
Ah! I think I misunderstood the examples. I thought I could use either the "Basic" code block or the "Org-Cite" code block. 
So if I want to use org-cite I should do something like this ?  

```
;; Set bibliography paths so they are the same.
(defvar my/bibs '("~/bib/references.bib"))

(use-package bibtex-actions)
(use-package oc-bibtex-actions
  :bind (("C-c b" . org-cite-insert)
         ("M-o" . org-open-at-point)
         :map minibuffer-local-map
         ("M-b" . bibtex-actions-insert-preset))
  :after (embark oc)
  :config
  (setq bibtex-actions-bibliography my/bibs
        org-cite-global-bibliography my/bibs))

;; Use consult-completing-read for enhanced interface.
(advice-add #'completing-read-multiple :override #'consult-completing-read-multiple)
```

Thanks for the quick reply!

**Reply by bdarcus** [reply]
Yes, but in retrospect (per my edit), try it without the `(use-package bibtex-actions)` line; it should still work.

**Reply by gwbrck** [reply]
without `(use-package bibtex-actions)` straight returns an error:

```
error: Could not find package oc-bibtex-actions. Updating recipe repositories: (org-elpa melpa gnu-elpa-mirror el-get emacsmirror-mirror) with ‘straight-pull-recipe-repositories’ may fix this
```
I think the problem is that the package name on melpa is different from the package name that should be loaded. Maybe this is an reasonable way (?) at least for straight.el: 

```
(use-package oc-bibtex-actions
      :straight (bibtex-actions)
      :bind (("C-c b" . org-cite-insert)
             ("M-o" . org-open-at-point)
...
```

**Reply by bdarcus** [reply]
Ah, yeah; if you're using straight to install also.

Does that work then?

**Reply by gwbrck** [reply]
how can I check if oc-bibtex-actions is loaded instead of Bibtex-actions? 🙈

**Reply by bdarcus** [reply]
If you're in org and you do `M-x org-cite-insert`, you should see the bibtex-actions UI (with the "References" prompt).

After inserting a citation, you should be able to run `embark-act` while point is on it and get the embark menu.

**Reply by gwbrck** [reply]
no it's not working... 
I also tried this: 

```
    (use-package bibtex-actions)
    (use-package oc-bibtex-actions
      :straight nil
      :bind (("C-c b" . org-cite-insert)
             ("M-o" . org-open-at-point)
             :map minibuffer-local-map
             ("M-b" . bibtex-actions-insert-preset))
```

**Reply by bdarcus** [reply]
I may have forgotten something. What about adding this to the `:config` section?
```elisp
  (setq org-cite-insert-processor 'oc-bibtex-actions
        org-cite-follow-processor 'oc-bibtex-actions
        org-cite-activate-processor 'basic))
```

**Reply by bdarcus** [reply]
If that doesn't work, how are you installing org (master) itself?

**Reply by gwbrck** [reply]
I install with straight and this recipe
`(org :type git :repo "https://code.orgmode.org/bzg/org-mode.git" :local-repo "org" :depth full :pre-build
(straight-recipes-org-elpa--build)
:build
(:not autoloads)
:files
(:defaults "lisp/*.el"
("etc/styles/" "etc/styles/*")))
`

**Reply by gwbrck** [reply]
I tried it with the following code in `:config` but it still doesn't work.
```
  (setq org-cite-insert-processor 'oc-bibtex-actions
        org-cite-follow-processor 'oc-bibtex-actions
        org-cite-activate-processor 'basic))
```
maybe straight.el isn't building the master version of org. But I see `M-x org-cite-insert`. So I assumed straight builds the master branch.

**Reply by bdarcus** [reply]
Can you clarify what you mean by "doesn't work"?

**Reply by gwbrck** [reply]
Sorry! I meant that I could not load oc-bibtex-actions properly with straight.el. 

I managed to load it like this:
```Elisp
  (use-package bibtex-actions)
  (use-package oc-bibtex-actions
    :straight nil
    :config
      (setq bibtex-actions-bibliography "~/Documents/myBib/main.bib")
      (setq org-cite-insert-processor 'oc-bibtex-actions
                org-cite-follow-processor 'oc-bibtex-actions
                org-cite-activate-processor 'basic))
```
But I think this is a bit quirky

**Reply by gwbrck** [reply]
I think it's because I'm using `straight-use-package-by-default`, which is an equivalent of `use-package-always-ensure`...

**Reply by bdarcus** [reply]
> I managed to load it like this:

So to be clear, now working fine for you?

**Reply by gwbrck** [reply]
Yes it works.

**Reply by mclearc** [reply]
Just FYI -- this should work as an entry in case you want to configure them differently:

```emacs-lisp
(use-package bibtex-actions
  :straight (:host github :repo "bdarcus/bibtex-actions" :includes oc-bibtex-actions))

(use-package oc-bibtex-actions
  :after (oc embark)
  :commands (oc-bibtex-actions-select-style oc-bibtex-actions-insert)
  :config
  (setq org-cite-insert-processor 'oc-bibtex-actions
        org-cite-follow-processor 'oc-bibtex-actions))
```
See [the recipe format](https://github.com/raxod502/straight.el#the-recipe-format), or just search for ":includes" on the straight readme.

**Comment by bdarcus** [comment, marked as answer]
Answer is here:

https://github.com/bdarcus/bibtex-actions/discussions/275#discussioncomment-1297146
